### ISSUE TITLE
Too much time wait the client to resume file download if the seeders goes online

### ISSUE THREAD

**Issue by mx5kevin** [HelloZeroNet/ZeroNet#2428]
### Step 1: Please describe your environment

  * ZeroNet version: 0.7.1 (rev4445)
  * Operating system: Windows 10
  * Web browser: Firefox 72.02
  * Tor status: not always
  * Opened port: no
  * Special configuration:

Used 2 different wifi netwotks with 2 different internet provider. 150KB/s upload speed. 3GB of total files in cc 500 MB/ file.1 week period and the  computers serval times goes off and on. And not all 2 PC always active at the same time.

### Step 2: Describe the problem:

Too much time wait the client to resume file download if the seeders goes online. Without the user interaction using the refresh button and shows example 2000 file left. Too much time passes when the seeder goes online and the client start to download the files.This largely increases the time it takes for the files to be fully downloaded. 5piece cc 500-600MB  total 3 GB file about 1 week downloaded. Which is a lot of time.

#### Steps to reproduce:

1. Use 2 PC with 2 different internet provider what restarted, and behave like any other PC in the internet.

#### Observed Results:

When the seeder goes online too much time passes when the client start refresh the current page and download the files. Without the auto refreshing the page files will not downloading.About hours the client does not download the files even if the seeder are goes online.

#### Expected Results:

Better detect it if the another client are online. About 1 week from 3GB of total files using 150KB/s upload speed it's a lot of time. And there was too much time to the 2 PC was the same time online and the client can download the file, but it didn't.

**Comment by HelloZeroNet** [comment]
Does the seeder/downloader computer has opened port or you using Tor network for the transfer?

**Comment by mx5kevin** [comment]
No all client the port are closed and Tor are in always mode. The page auto refreshing are too slow. The download starts when i manualy refresh the site. The problem  if the cilent does not refresh the files then there is no download. The transmission is working, the time wastage is too much when the cilent try to download the file. If i update it the zite manually the transmission always starts immediately. I have to wait too long for this automatically. It does not try to look for active seeders for too long.The files refreshing for too long inactive.Several times the two machines were active.And there was no transfer.But when I refreshed it by manualy then the transfer always started.The client always uploaded several 5 files at once used the Help distribute added optional files option.

**Comment by slrslr** [comment]
Yes, i would also welcome if i can make zeronet retry more frequently (numerous times i seen it is not downloading and when i click onsite seed button it resumed download and downloaded successfully) or if it is made the way that the seeder went online and its zeronet announce itself to other peers and peers resume downloading optional files without delay.

**Comment by HelloZeroNet** [comment]
Yeah it could be improved by starting the file download when the client detects a new peer for the optional file. Currently it only check every X minute if the has peers or not.
